lib/Exception/UnexpectedStatusCodeException.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace DCG\Cinema\Exception;
+
+use Throwable;
+
+class UnexpectedStatusCodeException extends \RuntimeException
+{
+    private $statusCode;
+
+    /**
+     * @param int $statusCode
+     * @param string|null $message
+     * @param int $code
+     * @param Throwable|null $previous
+     */
+    public function __construct($statusCode, $message = null, $code = 0, $previous = null)
+    {
+        $this->statusCode = $statusCode;
```

Comment:
Rather than storing statusCode directly, you could just use the parent exceptions getCode interface